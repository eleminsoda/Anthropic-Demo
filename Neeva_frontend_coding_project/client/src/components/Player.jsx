import React from "react";
import { useEffect } from "react";
import { useState } from "react";
import useFPS from "../utils/useFPS";
import FPSButtonGroup from "./FPSButtonGroup";
import Spinner from "react-bootstrap/Spinner";
import Button from "react-bootstrap/Button";


export default function Player(props) {
    const [isPaused, setIsPaused] = useState(false);
    const [fps, setFPS] = useState(30);
    const [isLoading, setIsLoading] = useState(true);
    const [currentImage, setCurrentImage] = useState(0);
    const [images, setImages] = useState([]);

    const pause = () => {
        setIsPaused(!isPaused);
    };

    useEffect(() => {
        // fetch all the links for images
        fetch("http://localhost:3001/video/1?offset=0&count=600")
            .then((res) => res.json())
            .then(
                (res) =>
                    new Promise((resolve) => {
                        resolve(res.frames);
                    })
            )
            .then((res) =>
            // fetch the real data for the images
                Promise.all(
                    res.map((link) =>
                        fetch("http://localhost:3001/" + link).then(
                            (res) => res.url
                        )
                    )
                )
            )
            .then((res) => {
                setImages(res);
                setIsLoading(false);
            });
    }, []);

    useFPS(
        () => {
            setCurrentImage(
                currentImage + 1 > images.length ? 0 : currentImage + 1
            );
        },
        isPaused ? null : fps
    );

    return (
        <div className="player">
            <div className="image-stack">
                {isLoading ? (
                    <Spinner
                        animation="border"
                        role="status"
                        className="spinner"
                    >
                        <span className="visually-hidden">Loading...</span>
                    </Spinner>
                ) : (
                    <div>
                        {images.map((img, index) => (
                            <img
                                alt="Dinner on the rooftop"
                                className={
                                    currentImage === index
                                        ? "active"
                                        : "inactive"
                                }
                                key={index}
                                src={img}
                            />
                        ))}

                        <Button
                            id="play-button"
                            onClick={pause}
                            variant="secondary"
                        >
                            <svg
                                xmlns="http://www.w3.org/2000/svg"
                                id="play-icon"
                                fill="currentColor"
                                viewBox="0 0 16 16"
                            >
                                <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM6.79 5.093A.5.5 0 0 0 6 5.5v5a.5.5 0 0 0 .79.407l3.5-2.5a.5.5 0 0 0 0-.814l-3.5-2.5z" />
                            </svg>
                        </Button>
                    </div>
                )}
            </div>

            <FPSButtonGroup callback={setFPS} />
        </div>
    );
}
