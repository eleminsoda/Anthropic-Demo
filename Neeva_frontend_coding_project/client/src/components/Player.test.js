import renderer from "react-test-renderer";
import { render, unmountComponentAtNode } from "react-dom";
import { expect, jest } from "@jest/globals";
import { act } from "react-dom/test-utils";
import Player from "./Player";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});


it("the player should render correctly", async () => {
    const fakeImageLinks = {
        frames: [],
    };
    jest.spyOn(global, "fetch").mockImplementation(() =>
        Promise.resolve({
            json: () => Promise.resolve(fakeImageLinks),
        })
    );

    // Use the asynchronous version of act to apply resolved promises
    await act(async () => {
        render(<Player />, container);
    });

    expect(container).toMatchSnapshot();

    // remove the mock to ensure tests are completely isolated
    global.fetch.mockRestore();
});
