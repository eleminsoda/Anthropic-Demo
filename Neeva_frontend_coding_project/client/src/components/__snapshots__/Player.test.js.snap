// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the player should render correctly 1`] = `
<div>
  <div
    class="player"
  >
    <div
      class="image-stack"
    >
      <div>
        <button
          class="btn btn-secondary"
          id="play-button"
          type="button"
        >
          <svg
            fill="currentColor"
            id="play-icon"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM6.79 5.093A.5.5 0 0 0 6 5.5v5a.5.5 0 0 0 .79.407l3.5-2.5a.5.5 0 0 0 0-.814l-3.5-2.5z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      aria-label="FPS"
      class="button-group btn-group"
      role="group"
    >
      <button
        class="btn btn-secondary"
        type="button"
      >
        10
      </button>
      <button
        class="btn btn-secondary"
        type="button"
      >
        30
      </button>
      <button
        class="btn btn-secondary"
        type="button"
      >
        60
      </button>
    </div>
  </div>
</div>
`;
